src/Symfony/Component/PasswordHasher/Hasher/UserPasswordHasher.php

```diff
@@ -27,36 +31,66 @@ public function __construct(PasswordHasherFactoryInterface $hasherFactory)
         $this->hasherFactory = $hasherFactory;
     }
 
-    public function hashPassword(UserInterface $user, string $plainPassword): string
```

Comment:
As the typehint is removed, you need to trigger the `TypeError` in the method when passing the wrong type.